### PR TITLE
Updating the API

### PR DESCRIPTION
**Pull request by CheezItMan**
Looking at this, I updated the API to a newer version of Rails & Ruby, and added Dan's Cors test and moved it toward Postgresql.

Still not done, but putting this PR up for review to date

Finished:
* Figure out what is wrong with checkin logic
* Update Rails version and fix security vulnerabilities
* Switch to Postgresql
* Tested deployment to:  https://frozen-woodland-27192.herokuapp.com

Yet to do: 
* Add travis CI
* Take out the assertions and move toward spec-style testing

### CODE REVIEW

**Comment by droberts-ada** [comment]
No Travis. We don't deploy this API, the students fork it and hack on it themselves. If they deploy they'll do it manually.

**Comment by droberts-ada** [comment]
Whoops I forgot we have a deployed demo of this! Ignore my earlier comments about Travis

File: travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+services:
+- postgresql
+before_script:
+- bin/rails db:create
+- bin/rails db:migrate
+deploy:
+  provider: heroku
+  api_key:
+    secure: TODO
+  app: videostoreconsumerapi
```
Comment (droberts-ada) [inline travis.yml]: You can run `travis setup heroku --force` and follow the prompts to set this up.

**Comment by CheezItMan** [comment]
@droberts-ada I've added a test to prevent the check-in bug from reoccuring.  It involved the fact that Checking out a movie left `returned` as `nil` and check-in required it to be `false`.  So I've got Rental to default returned to `false`.

**Pull request merged**